bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/api/Runtime.java

```diff
@@ -61,6 +54,37 @@ public static Runtime getCurrentRuntime() {
         return new Runtime(strand.scheduler);
     }
 
+    /**
+     * Invoke Object method asynchronously. This will schedule the function and block the strand.
+     *
+     * @param object     Object Value.
+     * @param methodName Name of the method.
+     * @param strandName Name for newly creating strand which is used to execute the function pointer. This is
+     *                   optional and can be null.
+     * @param metadata   Meta data of new strand.
+     * @param callback   Callback which will get notify once method execution done.
+     * @param isIsolated Ensures method can be called preserving isolation with no data race is possible for the
+     *                   mutable state freely reachable from the arguments that it passes to the function.
+     * @param properties Set of properties for strand
+     * @param returnType Expected return type of this method
+     * @param args       Ballerina function arguments.
+     * @return           {@link FutureValue} containing return value of executing this method.
+     */
+    public BFuture invokeMethodAsync(BObject object, String methodName, String strandName, StrandMetadata metadata,
+                                     boolean isIsolated, Callback callback, Map<String, Object> properties,
+                                     Type returnType, Object... args) {
+        if (object == null) {
+            throw new NullPointerException();
```

Comment:
I think the more suitable  exception would be `IllegalArgumentException` because we do not want the argument `object` to be null. Otherwise we can choose to use it an `Optional` argument as input: https://www.oracle.com/technical-resources/articles/java/java8-optional.html

Reply:
I think we could also have a test to cover this line.